{
  expect(callWithCurrentContinuation(new CallbackReceiver<Integer>(){
    public Integer receive(    Callback<Integer> continuation){
      continuation.handleResult(42);
      throw new AssertionError();
    }
  }
) == 42);
  expect(callWithCurrentContinuation(new CallbackReceiver<Integer>(){
    public Integer receive(    Callback<Integer> continuation){
      return 43;
    }
  }
) == 43);
  try {
    callWithCurrentContinuation(new CallbackReceiver<Integer>(){
      public Integer receive(      Callback<Integer> continuation){
        continuation.handleException(new MyException());
        throw new AssertionError();
      }
    }
);
    throw new AssertionError();
  }
 catch (  MyException e) {
    e.printStackTrace();
  }
  try {
    callWithCurrentContinuation(new CallbackReceiver<Integer>(){
      public Integer receive(      Callback<Integer> continuation) throws MyException {
        throw new MyException();
      }
    }
);
    throw new AssertionError();
  }
 catch (  MyException e) {
    e.printStackTrace();
  }
}
